### DISCUSSION TITLE
Kratos user identities and header information

### DISCUSSION THREAD

**Discussion by rezakazemi928** [ory/kratos#2065] [Q&A] [answered]
Hi,
I am writing a flask application with python and I was wondering, Can I get the last login users header information such as User-Agent and etc. I checked the /get/identities endpoint and I could not find any pieces of information such as user-agent??
Is it even possible to get this kind of information from Kratos ??

**Comment by joekrill** [comment, marked as answer]
I've been looking into this myself. This information would probably have to be stored for each _session_ - not necessarily at the identity level.

And as far as I can tell this information isn't currently saved anywhere at all. One option I considered was using a web hook to store the information externally (the web hook can include the original request headers). The problem is I don't think the session ID is available to the web hook, so I'm not sure how you would match up a particular web hook call with a specific session.

I created a feature request for something like this: #2091

**Reply by vinckr** [reply]
Hey @rezakazemi928 ,
I marked Joes response as an answer, I think there is nothing to add :-). 
Check out the linked issue, a great overview of what needs to be added to make this possible in Ory Kratos out of the box.
